src/components/input/math-input.js

```diff
@@ -255,36 +255,81 @@ const MathInput = React.createClass({
         });
     },
 
+    _forwardGlobalKeydown(e) {
+        if (e.keyCode === 13 || e.keyCode === 27) { // Enter, Esc
+            this.props.keypadElement.dismiss();
```

Comment:
Just to make sure I follow: by dismissing, we then blur, and thereby avoid forwarding the key-press event for an Enter or Escape down to MathQuill. Is that right?